query-service-impl/build.gradle.kts

```diff
@@ -41,18 +41,38 @@ dependencies {
   implementation("org.hypertrace.core.attribute.service:attribute-projection-registry:0.12.3")
   implementation("org.hypertrace.core.attribute.service:caching-attribute-service-client:0.12.3")
   implementation("com.google.inject:guice:5.0.1")
-  implementation("org.apache.pinot:pinot-java-client:0.6.0") {
+  implementation("org.apache.pinot:pinot-java-client:0.12.0") {
     // We want to use log4j2 impl so exclude the log4j binding of slf4j
     exclude("org.slf4j", "slf4j-log4j12")
   }
+  constraints {
+    implementation("com.fasterxml.jackson.core:jackson-databind:2.13.4.2") {
+      because("version 2.12.7.1 has a vulnerability https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038424")
+    }
+    implementation("org.apache.calcite:calcite-core:1.32.0") {
+      because("version 1.29.0 has a vulnerability https://security.snyk.io/vuln/SNYK-JAVA-ORGAPACHECALCITE-3021143")
+    }
+    implementation("org.apache.calcite.avatica:avatica-core:1.22.0") {
+      because("version 1.20.0 has a vulnerability https://security.snyk.io/vuln/SNYK-JAVA-ORGAPACHECALCITEAVATICA-2961770")
+    }
+    implementation("commons-httpclient:commons-httpclient:3.1-jenkins-3") {
+      because("version 3.1 has a vulnerability https://security.snyk.io/vuln/SNYK-JAVA-COMMONSHTTPCLIENT-30083")
+    }
+    implementation("io.netty:netty-codec-http:4.1.71.Final") {
+      because("version 4.1.60.Final has a vulnerability https://security.snyk.io/vuln/SNYK-JAVA-IONETTY-2314893")
+    }
+    implementation("org.webjars:swagger-ui:4.1.3") {
```

Comment:
Where are all these coming from? can we try and fix or not use the upstream, there's no reason we should have a swagger ui dep anywhere in the tree.

Reply:
introduced by org.apache.pinot:pinot-java-client@0.12.0 
```
  No upgrade or patch available
  ✗ User Interface (UI) Misrepresentation of Critical Information [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-ORGWEBJARS-2314887] in org.webjars:swagger-ui@3.23.11
    introduced by org.apache.pinot:pinot-java-client@0.12.0 > org.apache.pinot:pinot-common@0.12.0 > org.webjars:swagger-ui@3.23.11
  This issue was fixed in versions: 4.1.3
  ✗ Insecure Defaults [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-ORGWEBJARS-575003] in org.webjars:swagger-ui@3.23.11
    introduced by org.apache.pinot:pinot-java-client@0.12.0 > org.apache.pinot:pinot-common@0.12.0 > org.webjars:swagger-ui@3.23.11
  This issue was fixed in versions: 3.26.1
```

Reply:
pinot's client brings in all this?! that's disturbing. Any chance we can raise an issue with the upstream on that? This is a huge waste, bloat and bug/vulnerability vector.

Anyway, not a blocker here.

What I would be careful of however is pinot making breaking changes in their client since it's a pre-1.0 client and doesn't match our server version. As long as we e2e tested, should be ok though.

Reply:
Moving to version 0.10.0 removed all these constraints. Marking this as resolved.